File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/randomFactory/RandomFactory.java
package com.research.pocketKeanu.randomFactory;

import com.research.pocketKeanu.abstractTypes.DoubleLike;
import com.research.pocketKeanu.abstractTypes.IntLike;

public interface RandomFactory<D extends DoubleLike<D>, I extends IntLike<I>> extends DoubleLikeRandomFactory<D>, IntLikeRandomFactory<I> {
}

File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/randomFactory/RandomFactory.kt
package com.research.pocketKeanu.randomFactory

import com.research.pocketKeanu.abstractTypes.DoubleLike
import com.research.pocketKeanu.abstractTypes.IntLike

interface RandomFactory<D : DoubleLike<D>, I : IntLike<I>> : ContinuousRandomFactory<D>, DiscreteRandomFactory<I>
